### DISCUSSION TITLE
Sort table

### DISCUSSION THREAD

**Discussion by rafikApp** [Budibase/budibase#9446] [Help] [answered]
Hi everyone.
when I want to sort a table by clicking on a column, the table empties.
Any solution please ?

@melohagan 

https://user-images.githubusercontent.com/106697208/214756931-09477c81-c68b-473a-9a77-5bd4beba3205.mp4

**Comment by melohagan** [comment, marked as answer]
Hey @rafikApp 

What is your datasource, what are the column data-types, what does your component tree look like, what does the pagination and sorting settings look like?

**Reply by rafikApp** [reply]
Hi @melohagan 

I use budibase datasource, 
Date et heure type = Date/Time
Zone Type = Formula
Entreprise type = Relationship
Intervenant type = Formula
statut type = Options


![tree](https://user-images.githubusercontent.com/106697208/214819649-fd0c647f-fdc8-4032-8ba4-813e48016edb.PNG)
![data](https://user-images.githubusercontent.com/106697208/214819661-03a41f6f-a26c-4dca-a5ad-e5d69629c939.PNG)
![table](https://user-images.githubusercontent.com/106697208/214819679-d20eff2c-2f5c-4948-b54f-5703929fd6c2.PNG)

**Reply by melohagan** [reply]
Thanks for that!

Would you also be able to show me the contents of the two Formula columns please? (Zone, and Intervenant type)

**Reply by melohagan** [reply]
Thank you, and finally what version of Budibase is your app using?

Also, do you see any errors in the Console or Network tab after you click to sort and no rows are returned?

**Reply by rafikApp** [reply]
Current version
2.0.12

No there's no errors on the console.
about the network tab
![network](https://user-images.githubusercontent.com/106697208/214825190-847995ef-2491-4700-b49d-7fae25fe2932.PNG)

**Reply by melohagan** [reply]
Ah, I'm pretty sure that bug has been fixed. Please upgrade to the latest version, currently: **2.2.22**

**Reply by melohagan** [reply]
Hey @rafikApp I believe I may know what the problem is. In the **Data** section edit the **FORCEE** column. Remove the illegal character and save column.

We made changes fairly recently to only allow the standard a-z, 0-9 and _ within column names to get around some issues that were being caused.

**Reply by rafikApp** [reply]
Yes exactly, the problem is solved.
Thank you @melohagan
